provider/internal/java/dependency.go

```diff
@@ -97,52 +96,65 @@ func getMavenLocalRepoPath(mvnSettingsFile string) string {
 	return string(outb.String())
 }
 
-func (p *javaServiceClient) GetDependencyFallback(ctx context.Context) (map[uri.URI][]*provider.Dep, error) {
-	pomDependencyQuery := "//dependencies/dependency/*"
-	path := p.findPom()
-	file := uri.File(path)
-
-	absPath, err := filepath.Abs(path)
-	if err != nil {
-		return nil, err
-	}
+func (p *javaServiceClient) GetDependenciesFallback(ctx context.Context, location string) (map[uri.URI][]*provider.Dep, error) {
+	deps := []*provider.Dep{}
 
-	f, err := os.Open(absPath)
-	if err != nil {
-		return nil, err
-	}
-	doc, err := xmlquery.Parse(f)
-	if err != nil {
-		return nil, err
+	path := p.findPom()
+	if location != "" {
+		path = location
 	}
-	list, err := xmlquery.QueryAll(doc, pomDependencyQuery)
+	pom, err := gopom.Parse(path)
 	if err != nil {
 		return nil, err
 	}
-	deps := []*provider.Dep{}
-	dep := &provider.Dep{}
-	// TODO this is comedically janky
-	for _, node := range list {
-		if node.Data == "groupId" {
-			if dep.Name != "" {
-				deps = append(deps, dep)
-				dep = &provider.Dep{}
+
+	// have to get both <dependencies> and <dependencyManagement> dependencies (if present)
+	var pomDeps []gopom.Dependency
+	if pom.Dependencies != nil {
+		pomDeps = append(pomDeps, *pom.Dependencies...)
+	}
+	if pom.DependencyManagement != nil {
+		pomDeps = append(pomDeps, *pom.DependencyManagement.Dependencies...)
+	}
+
+	// add each dependency found
+	for _, d := range pomDeps {
+		dep := provider.Dep{}
+		dep.Name = fmt.Sprintf("%s.%s", *d.GroupID, *d.ArtifactID)
```

Comment:
Are there times when GroupID or ArtifactID could be nil, this will panic if they can be.

Reply:
Not with a well formed pom - if it can be compiled, dependencies will have at least groupId and artifactId